src/wast-parser.cc

```diff
@@ -1488,6 +1497,96 @@ Result WastParser::ParsePlainInstr(std::unique_ptr<Expr>* out_expr) {
   return Result::Ok;
 }
 
+// Simd
+// The expected V128 const lists is:
+// i32 0xXXXXXXXX 0xXXXXXXXX 0xXXXXXXXX 0xXXXXXXXX
+Result WastParser::ParseSimdConstList(ConstVector* consts) {
+  WABT_TRACE(ParseSimdConstList);
+  Literal literal;
+
+  // Check input const type.
+  if (!PeekMatch(TokenType::ValueType))
+    return Result::Error;
+
+  Type in_type = Consume().type();
+
+  switch (in_type) {
+    case Type::I32: {
+      // Meet expected "i32" token. start parse 4 i32 consts
+      for(int i=0; i<4; i++) {
+        Const const_;
+        const_.loc = GetLocation();
+
+        // Expected one 0xXXXXXXXX number
+        if (!PeekMatch(TokenType::Nat))
+          return ErrorExpected({"an Nat literal"}, "123");
+
+        Literal literal = Consume().literal();
+
+        string_view sv = literal.text;
+        const char* s = sv.begin();
+        const char* end = sv.end();
+        Result result;
+
+        const_.type = Type::I32;
+
+        result =
+            ParseInt32(s, end, &(const_.u32), ParseIntType::SignedAndUnsigned);
+
+        if (Failed(result)) {
+          Error(const_.loc, "invalid literal \"%s\"", literal.text.c_str());
+        }
+
+        consts->push_back(const_);
+      }
+      break;
+    }
+
+    default:
+      assert(!"ParseSimdConstListcalled with invalid input const type");
+      return Result::Error;
+  }
+
+  return Result::Ok;
+}
+
+Result WastParser::ParseSimdConst(Const* const_) {
```

Comment:
I think it would be better if this was called from `ParseConst` so we could share places where that is used currently.